src/Form/Form.js

```diff
@@ -442,6 +443,30 @@ class Form {
                 return
             }
         }
+
+        // if we have an unknown input, and only one non-hidden password input, we can assume the unknown input is the username
+        // In that case we move the inputs to the correct category
+        if (this.canCategorizeUnknownUsername()) {
+            const hasUsername = [...this.inputs.all].some(input => getInputSubtype(input) === 'username')
+            if (this.inputs.unknown.size === 1 && this.isLogin && !hasUsername) {
+                const unknownInput = [...this.inputs.unknown][0]
+
+                const passwordInputs = [...this.inputs.credentials].filter(
+                    (/** @type {HTMLInputElement} */ input) => {
+                        const isPassword = getInputSubtype(input) === 'password'
+                        const isVisible = input.style.display !== 'none' && input.style.visibility !== 'hidden'
```

Comment:
Ignore this, I will add a bit better visibility logic :D For now the crude approach works.

Reply:
Why do we care for password visibility?

Reply:
This is specific to Bank of America, where there are two password fields that are decorated with `credentials.password.current`. However, only one of the fields are visible. If we don't check this, then the categorization of the username will fail because in that case there will be two password fields.

I think it's reasonable enough to restrict the password based categorization with visible password fields (in this case with `display` property, since that's what BoA does to hide the field), so that we don't miss out on such cases. We can also do an extensive visibility check, but I am only intending to fix the BoA site's problems here at the moment.

Reply:
Got it. I'm not convinced we should limit to 1 password field. What's the downside if there are 2 or more?

Reply:
I saw some more input classifier tests fail. I am going to check it out in a bit more, if those are legit or not. Here we are limiting the check to login forms too, where double password fields are less likely.

Reply:
Okay! I broadened the check to any number of inputs (more than 0), and it seems like we don't introduce any other regression (apart from the chase.com). Also added further checks to bail out of the form is more complex than we analyse.

Probably the Form analyzer should be able to do better here (categorizing `simplii_login.html` form as a "login" form e.g right now), but I'd say that's a bit larger scope and needs a bit of back and forth.